src/package-linker.js

```diff
@@ -400,7 +406,7 @@ export default class PackageLinker {
     for (const [dest, {pkg, isDirectRequire}] of flatTree) {
       const {name} = pkg;
 
-      if (!linksToCreate.has(name) || isDirectRequire) {
+      if (isDirectRequire || (this.topLevelBinLinking && !linksToCreate.has(name))) {
```

Comment:
We may wanna go one step back and prevent these from getting added to `linksToCreate` at all may be?

Reply:
Oh, NVM. I'm just being stupid lol